### ISSUE TITLE
When used with Containerise, tabs open twice and get focus

### ISSUE THREAD

**Issue by shvchk** [stoically/temporary-containers#15]
When used with [Containerise](https://github.com/kintesh/containerise), links open twice (second tab with the same links is opened and almost always instantly closed, but sometimes it stays). This slows down page loading and also makes almost every link appear in history twice. Also, if link was opened in background tab, that tab gets focused, so you always have to switch back to the parent tab.

**Comment by stoically** [comment]
Hi, could you provide informations on how to exactly reproduce this bug?

* Settings in Containerise
* Settings in Temporary Containers
* What and how to click

Thanks.

**Comment by dakka2** [comment]
Seems to happen in automatic mode ctrl left clicking links. Additionally happens opening new blank tabs (opens two blank tabs)

**Comment by stoically** [comment]
When I Ctrl+LeftClick a link in a tmpcontainer that is bound to a permanent container using Containerise, it correctly opens only one Tab in the assigned permanent container.

Pressing Ctrl+T also only opens one New Tab in a temporary container for me.

Unfortunately I can't do anything without exact steps how to reproduce. Other helpful informations beside the ones asked in my previous informations would be the versions of Firefox and the Addons and which Operating System.

If it's not always reproducible I'd need a debug log for the case in which it happens. It can be obtained by going to `about:debugging`, Enable Debugging, Clicking on "Debug" next to "Temporary Containers" and executing `log.DEBUG = true;` in the DevTools Console. Then Copy&(No)Paste the whole log for the case in which it happens.

**Comment by schnittchen** [comment]
I have the same problem. I am not using the containerized extension, but IIRC the feature has been integrated by now. The behaviour is linked with the "Always open in <container>" feature.

Recently, this has gotten worse: When following a link, a new tab would open and shortly after close again.

Right now this prevents me from using this extension, although I love the idea.

**Comment by stoically** [comment]
In general, if you upgraded to Firefox58 (or used a version > 57 before already), those versions have a bug that I was able to reproduce, see #20 - unfortunately the published version 0.20 introduced a regression and I had to deactivate. This is happens because of [race-conditions with MAC](https://github.com/mozilla/multi-account-containers/issues/929#issuecomment-348376375) and is really hard to fix in a sane way. I'll try to get a workaround ready and will look into opening a PullRequest for MAC and introduce some sort of API.

@schnittchen If you can reproduce the unexpected closing of Tabs bug I'd appreciate if you could open a new Issue with browser/addon versions+settings and how to reproduce.

**Comment by stoically** [comment]
Just released a fix for #20 

@shvchk Could you please test if the problem still exists with the latest version 0.23? Thanks.

**Comment by stoically** [comment]
@dakka2 Could you also test if the problem still exists for you?

Also, if you guys could vote for [this PR](https://github.com/mozilla/multi-account-containers/pull/1095), that'd be great, thanks!

**Comment by stoically** [comment]
To fully support Containerise we would need them to implement something similar to https://github.com/mozilla/multi-account-containers/pull/1095. After #29 is done I will look into opening a Pull request with Containerise.

**Comment by Gitoffthelawn** [comment]
I don't use Containerise, but am experiencing a similar issue.  I'll try to narrow down what might be causing it as soon as I have a little time.

**Comment by stoically** [comment]
@Gitoffthelawn Would be great if you could open a new Issue if you find the time to reproduce, since this Issue should be kept Containerise-specific.

Regarding Containerise: There's a pending Pull request now https://github.com/kintesh/containerise/pull/18 - compatibility can finally be achieved if that gets merged.

**Comment by Gitoffthelawn** [comment]
@stoically Will do.

**Comment by Gitoffthelawn** [comment]
Done.  See #106

**Comment by stoically** [comment]
Published version 0.90 with containerise support
